iocore/net/quic/QUICStreamFactory.h

```diff
@@ -0,0 +1,41 @@
+/** @file
+ *
+ *  A brief file description
+ *
+ *  @section license License
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#pragma once
+
+#include "QUICTypes.h"
+
+class QUICStreamVConnection;
+
+// PS: this class function should not static because of  THREAD_ALLOC and THREAD_FREE
+class QUICStreamFactory
+{
+public:
+  // stream. negative == true means this function called by incoming frame. If the stream_id indicate an unidirectional stream,
+  // then create a send only stream, otherwise (negtive == false) create a send only stream.
+  QUICStreamVConnection *create(QUICRTTProvider *rtt_provider, QUICConnectionInfoProvider *cinfo, QUICStreamId sid,
```

Comment:
We should probably be able to pass `rtt_provider` and `cinfo` to the constructor, then we don't need to pass them on every `create()`.